lib/matplotlib/tests/test_image.py

```diff
@@ -757,6 +756,41 @@ def test_imshow_endianess():
     ax2.imshow(Z.astype('>f8'), **kwargs)
 
 
+@image_comparison(baseline_images=['imshow_masked_interpolation'],
+                  remove_text=True, style='default')
+def test_imshow_masked_interpolation():
+
+    cm = copy(plt.get_cmap('viridis'))
+    cm.set_over('r')
+    cm.set_under('b')
+    cm.set_bad('k')
+
+    N = 20
+    n = colors.Normalize(vmin=0, vmax=N*N-1)
+
+    # data = np.random.random((N, N))*N*N
+    data = np.arange(N*N, dtype='float').reshape(N, N)
+
+    data[5, 5] = -1
+
+    data[15, 5] = 1e5
+
+    # data[3, 3] = np.nan
+
+    data[15, 15] = np.inf
+
+    mask = np.zeros_like(data).astype('bool')
+    mask[5, 15] = True
+
+    data = np.ma.masked_array(data, mask)
+
+    fig, ax_grid = plt.subplots(3, 6)
+    for interp, ax in zip(sorted(mimage._interpd_), ax_grid.ravel()):
+        ax.set_title(interp)
```

Comment:
I don't think there's any point adding a title if `remove_text` is `True` in the decorator. Otherwise this patch looks good to me.